packages/config/eslint-preset.js

```diff
@@ -1,40 +1,148 @@
+            },
+        },
     ],
-  },
-  ignorePatterns: ["**/*.stories.tsx", "**/dist", "node_modules", "apps/docs"],
-}
+
+    parser: "@typescript-eslint/parser",
+    parserOptions: {
+        project: [path.resolve(__dirname, "tsconfig.json"), "./packages/*/tsconfig.json"],
+        sourceType: "module",
+        tsconfigRootDir: __dirname,
+    },
+
+    plugins: ["@typescript-eslint"],
+
+    root: true,
+
+    rules: {
+        "@typescript-eslint/array-type": ["error", { default: "generic" }],
+        "@typescript-eslint/await-thenable": "error",
+        "@typescript-eslint/consistent-type-imports": ["error", { fixStyle: "inline-type-imports" }],
+        "@typescript-eslint/explicit-member-accessibility": "error",
+        "@typescript-eslint/method-signature-style": ["error", "method"],
+        "@typescript-eslint/no-meaningless-void-operator": "error",
+        "@typescript-eslint/no-non-null-asserted-nullish-coalescing": "error",
+        "@typescript-eslint/no-non-null-assertion": "error",
+        "@typescript-eslint/no-unnecessary-condition": "error",
+        "@typescript-eslint/no-unnecessary-type-arguments": "error",
+        "@typescript-eslint/no-unnecessary-type-assertion": "error",
+        "@typescript-eslint/no-unused-vars": [
+            "error",
+            {
+                argsIgnorePattern: "^_",
+                caughtErrorsIgnorePattern: "^_",
+                varsIgnorePattern: "^_",
+            },
+        ],
+        "@typescript-eslint/no-var-requires": "off",
+        "@typescript-eslint/prefer-optional-chain": "error",
+        "@typescript-eslint/prefer-readonly": "error",
+        "@typescript-eslint/promise-function-async": "error",
+        "@typescript-eslint/switch-exhaustiveness-check": "error",
+        "arrow-body-style": ["error", "as-needed"],
+        "consistent-return": "error",
+        "curly": "error",
+        "eqeqeq": ["error", "smart"],
+        "no-constant-binary-expression": "error",
+        "no-constructor-return": "error",
+        "no-else-return": "error",
+        "no-implicit-coercion": "error",
+        "no-lonely-if": "error",
+        "no-multi-assign": "error",
+        "no-multi-spaces": "error",
+        "no-new-native-nonconstructor": "error",
+        "no-param-reassign": "error",
+        "no-self-compare": "error",
+        "no-unneeded-ternary": "error",
+        "no-unused-private-class-members": "error",
+        "no-useless-concat": "error",
+        "no-useless-rename": "error",
+        "no-useless-return": "error",
+        "no-var": "error",
+        "object-curly-newline": [
+            "error",
+            {
+                ObjectExpression: {
+                    consistent: true,
+                    minProperties: 4,
+                    multiline: true,
+                },
+            },
+        ],
+        "object-shorthand": "error",
+        "padding-line-between-statements": [
```

Comment:
Personally not a fan of this, leads to a lot of empty lines in code that can be very concise otherwise